### ISSUE TITLE
connection seems to fail, despite what the client thinks

### ISSUE THREAD

**Issue by ShroomiePerson** [profanity-im/profanity#1951]
## Expected Behavior
when i type in connect and insert my password, i want it to connect. profanity thinks it's connected, but i can't message anyone and the client says `connection lost` anyways after some waiting and imputing anything, be it message or command. sometimes profanity _does_ connect, but this is in the minority of connection attempts. also notably, i think this is partially a server-side issue, as i recently moved servers and only started experiencing this issue when i did that.

## Steps to Reproduce (for bugs)
1. open profanity
2. type /connect
3. input your password
4. have profanity hallucinate your connection if you're unlucky (and you're very likely to be unlucky.)

## Context
i think it's pretty obvious _why_ i would want to be able to use my preferred client, so i'll tell a different but still important piece of context: this is an issue that only _i_ seem to be experiencing, so instead of asking somebody to fix a bug for me, i'm hoping  that someone can help me find more symptoms and diagnose the issue. 

## Environment
`profanity -v

Profanity, version 0.14.0
Copyright (C) 2012 - 2019 James Booth boothj5web@gmail.com.
Copyright (C) 2019 - 2023 Michael Vetter jubalh@iodoru.org.
License GPLv3+: GNU GPL version 3 or later https://www.gnu.org/licenses/gpl.html

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Build information:
XMPP library: libstrophe
Desktop notification support: Enabled
OTR support: Enabled (libotr 4.1.1)
PGP support: Enabled (libgpgme 1.23.2)
OMEMO support: Enabled
C plugins: Enabled
Python plugins: Disabled
GTK icons/clipboard: Disabled
GDK Pixbuf: Disabled`

i use a 2019 intel iMac running Mac OSX 14.2.1 and use the terminal emulator WezTerm. i don't know what glib is.

**Comment by jubalh** [comment]
Well... profanity connects for everyone else.
Does any other client connect with your new account/server?

**Comment by ShroomiePerson** [comment]
yep! though, they have issues loading OMEMO encrypted messages sent to and from me while using profanity.

**Comment by ShroomiePerson** [comment]
oh, i recently changed operating systems (i'm still running off of an imac though), and transferred some of my config files, and the problem still persists

**Comment by allora** [comment]
Similar for me, profanity as of some recent version no long logs into my xmpp server, while basically every other client does. 

It fails to properly use ssl stuff:
SCRAM-SHA-1 and SCRAM-SHA-1 Plus both fail to authenticate properly, ending in a SASL failure and disconnect from the server.

Server reports malformed request when the client responds to the challenge request

**Comment by jubalh** [comment]
And which libstrophe version are you running?

**Comment by sjaeckel** [comment]
Must be 0.13.0 since there was no SCRAM PLUS before.

I will create a patch release later today.

**Comment by allora** [comment]
latest on arch, which is 0.13.0-2

**Comment by sjaeckel** [comment]
BTW I'm not sure whether what @allora sees has something to do with the initial issue reported.

@allora your issue should be fixed with libstrophe 0.13.1

@ShroomiePerson can you provide logs please?

To do so start profanity via `profanity -l DEBUG`. The location of the logfile is printed to the console window when profanity started up.

You can also join the profanity MUC if you have any questions.

**Comment by allora** [comment]
@sjaeckel just to follow up, building profanity against this version of libstrophe went fine. This isnt yet updated in the arch repos though, so it'll probably be broken there till the packages in the extras repo get updated.

**Comment by jubalh** [comment]
Alpine, Gentoo, openSUSE, Termux, Ravenports have it updated (or update pending) already. Arch..slow as usual ;)

**Comment by ShroomiePerson** [comment]
@sjaeckel like, the logs after i connect?

**Comment by ShroomiePerson** [comment]
> @sjaeckel like, the logs after i connect?

well i mean, are the logs something that are made and completed when i open profanity, or are they something that update as i use profanity, and if it's the latter, what should i do before i quit profanity and upload the files?

**Comment by jubalh** [comment]
Just run the command as described and provide the log files.

**Comment by ShroomiePerson** [comment]
[profanity.log](https://github.com/profanity-im/profanity/files/14411206/profanity.log)
here's your thing

**Comment by Gustry** [comment]
I think I have the same issue.

On Ubuntu 24.04, it's still libstrophe 0.13.0 today.

I'm going to try to install manually.

**Comment by jubalh** [comment]
AFAIK profanity connects fine everywhere now ;)
